homeassistant/components/rejseplanen/sensor.py

```diff
@@ -6,252 +6,570 @@
+            _get_departure_attributes(departures, 0)
+            if departures
+            else _get_empty_departure_attributes()
+        ),
+    ),
+    RejseplanenSensorEntityDescription(
+        key="departures",
+        translation_key="departures",
+        value_fn=lambda departures: len(_get_current_departures(departures)),
+        attr_fn=_get_departures_list_attributes,
+    ),
+    RejseplanenSensorEntityDescription(
+        key="next_line",
+        translation_key="next_line",
+        value_fn=lambda departures: (
+            _get_current_departures(departures)[0].name
+            if _get_current_departures(departures)
+            else None
+        ),
+    ),
+    RejseplanenSensorEntityDescription(
+        key="next_direction",
+        translation_key="next_direction",
+        value_fn=lambda departures: (
+            _get_current_departures(departures)[0].direction
+            if _get_current_departures(departures)
+            else None
+        ),
+    ),
+    RejseplanenSensorEntityDescription(
+        key="next_track",
+        translation_key="next_track",
+        value_fn=lambda departures: (
+            _get_current_departures(departures)[0].rtTrack
+            or _get_current_departures(departures)[0].track
+            if _get_current_departures(departures)
+            else None
+        ),
+    ),
+)
 
 
-def setup_platform(
+async def async_setup_entry(
     hass: HomeAssistant,
-    config: ConfigType,
-    add_devices: AddEntitiesCallback,
-    discovery_info: DiscoveryInfoType | None = None,
+    config_entry: RejseplanenConfigEntry,
+    async_add_entities: AddConfigEntryEntitiesCallback,
 ) -> None:
-    """Set up the Rejseplanen transport sensor."""
-    name = config[CONF_NAME]
-    stop_id = config[CONF_STOP_ID]
-    route = config.get(CONF_ROUTE)
-    direction = config[CONF_DIRECTION]
-    departure_type = config[CONF_DEPARTURE_TYPE]
-
-    data = PublicTransportData(stop_id, route, direction, departure_type)
-    add_devices(
-        [RejseplanenTransportSensor(data, stop_id, route, direction, name)], True
+    """Set up the Rejseplanen sensors."""
+    coordinator = config_entry.runtime_data
+    entities: list[SensorEntity] = []
+    device_registry = dr.async_get(hass)
+
+    # Add service-level diagnostic sensors (associated with main service device)
+    entities.extend(
+        [
+            RejseplanenDiagnosticSensor(
+                coordinator=coordinator,
+                config_entry=config_entry,
+                key="last_update_time",
+                name="Last update time",
+                device_class=SensorDeviceClass.TIMESTAMP,
+            ),
+            RejseplanenDiagnosticSensor(
+                coordinator=coordinator,
+                config_entry=config_entry,
+                key="update_interval",
+                name="Update interval",
+                native_unit_of_measurement="s",
+            ),
+        ]
     )
```

Comment:
1. last update time is already stored per entity, no need to have an entity
2. The update interval is static, we don't want static information in the state machine